src/guigame.c

```diff
@@ -1295,6 +1312,102 @@ static void guiGameLoadPadMacroConfig(config_set_t *configSet, config_set_t *con
 }
 #endif
 
+//OSD
+
+static int guiGameOSDLanguageUpdater(int modified)
+{
+    int previousSource = gOSDLanguageSource;
+    diaGetInt(diaOSDConfig, OSD_LANGUAGE_SOURCE, &gOSDLanguageSource);
+
+    // update GUI to display per-game or global settings if changed
+    if (previousSource != gOSDLanguageSource && gOSDLanguageSource == SETTINGS_GLOBAL) {
+        config_set_t *configSet = gameMenuLoadConfig(diaOSDConfig);
+        configRemoveKey(configSet, CONFIG_ITEM_OSDLNG_SOURCE);
+        guiGameLoadOSDLanguageConfig(configSet, configGetByType(CONFIG_GAME));
+    } else if (previousSource != gOSDLanguageSource && gOSDLanguageSource == SETTINGS_PERGAME) {
+        config_set_t *configSet = gameMenuLoadConfig(diaOSDConfig);
+        configSetInt(configSet, CONFIG_ITEM_OSDLNG_SOURCE, gOSDLanguageSource);
+        guiGameLoadOSDLanguageConfig(configSet, configGetByType(CONFIG_GAME));
+    }
+
+    int LANGUAGE, ENABLE;
+    diaGetInt(diaOSDConfig, OSD_LANGUAGE_ENABLE, &ENABLE);
+    diaGetInt(diaOSDConfig, OSD_LANGUAGE_VALUE, &LANGUAGE);
+    gOSDLanguageEnable = ENABLE;
+    gOSDLanguageValue = LANGUAGE;
+
+    return 0;
+}
+
+void guiGameShowOSDLanguageConfig(int forceGlobal)
+{
+    const char *Lngs[] = {_l(_STR_LANGUAGE_JAPANESE), _l(_STR_LANGUAGE_ENGLISH), _l(_STR_LANGUAGE_FRENCH), _l(_STR_LANGUAGE_SPANISH), _l(_STR_LANGUAGE_GERMAN), _l(_STR_LANGUAGE_ITALIAN), _l(_STR_LANGUAGE_DUTCH), _l(_STR_LANGUAGE_PORTUGUESE), _l(_STR_LANGUAGE_RUSSIAN), _l(_STR_LANGUAGE_KOREAN), _l(_STR_LANGUAGE_TRAD_CHINESE), _l(_STR_LANGUAGE_SIMPL_CHINESE), NULL};
+    const char *sources[] = {_l(_STR_GLOBAL_SETTINGS), _l(_STR_PERGAME_SETTINGS), NULL};
+    diaSetEnum(diaOSDConfig, OSD_LANGUAGE_VALUE, Lngs);
+    diaSetEnum(diaOSDConfig, OSD_LANGUAGE_SOURCE, sources);
+    forceGlobalOSDLanguage = forceGlobal;
+    diaSetEnabled(diaOSDConfig, OSD_LANGUAGE_SOURCE, !forceGlobalOSDLanguage);
+
+    if (forceGlobalOSDLanguage) {
+        guiGameLoadOSDLanguageConfig(NULL, configGetByType(CONFIG_GAME));
+    }
+
+
+    int result = -1;
+    while (result != 0) {
+        result = diaExecuteDialog(diaOSDConfig, result, 1, &guiGameOSDLanguageUpdater);
+
+        if (result == UIID_BTN_OK)
+            break;
+    }
+}
+static int guiGameSaveOSDLanguageGameConfig(config_set_t *configSet, int result)
+{
+    if (gOSDLanguageSource == SETTINGS_PERGAME) {
+        result = configSetInt(configSet, CONFIG_ITEM_OSDLNG_SOURCE, gOSDLanguageSource);
+        result = configSetInt(configSet, CONFIG_ITEM_OSDLNG_ENABLE, gOSDLanguageEnable);
```

Comment:
Check the value of result. If result is set such as there is an error, don't do additional `configSetInt` calls